src/stackdriver-nozzle/firehose/client.go

```diff
@@ -27,17 +29,20 @@ func NewClient(apiAddress, username, password string, skipSSLValidation bool) Cl
 			ApiAddress:        apiAddress,
 			Username:          username,
 			Password:          password,
-			SkipSslValidation: skipSSLValidation}}
+			SkipSslValidation: skipSSLValidation},
+		nil}
 }
 
-func (c *client) StartListening(fh FirehoseHandler) error {
-	cfConfig := &cfclient.Config{
-		ApiAddress:        c.cfConfig.ApiAddress,
-		Username:          c.cfConfig.Username,
-		Password:          c.cfConfig.Password,
-		SkipSslValidation: c.cfConfig.SkipSslValidation}
-	cfClient := cfclient.NewClient(cfConfig)
+func (c *client) EnsureCfClient() *cfclient.Client {
```

Comment:
We don't, not sure if there is a benefit here. I'll move it into the New of firehose.Client